src/snapshot.js

```diff
@@ -93,6 +118,15 @@ export function restorePlayerSnapshot(player, snapshotObject) {
 
   const suppressedRemoteTracks = snapshotObject.suppressedRemoteTracks;
   const suppressedTracks = snapshotObject.suppressedTracks;
+
+  // ensures that the iOS TextTrackList 'change' listener added in getPlayerSnapshot()
+  // doesn't persist into main content
+  if (videojs.browser.IS_IOS) {
+    const tracks = player.textTracks();
+
+    tracks.off('change', snapshotObject.trackChangeHandler);
```

Comment:
This should also be cleaned up on contentupdate events in the reset handler: https://github.com/videojs/videojs-contrib-ads/blob/master/src/plugin.js#L193